lib/modules/EyeglassModules.js

```diff
@@ -253,39 +254,57 @@ function canAccessModule(name, origin) {
     return true;
   }
 
-  // find the nearest package for the origin
-  var pkg = packageUtils.findNearestPackage(origin);
+  var canAccessFrom = function canAccessFrom(origin) {
+    // find the nearest package for the origin
+    var pkg = packageUtils.findNearestPackage(origin);
 
-  var cache = this.cache.access;
-  cache[name] = cache[name] || {};
+    var cache = this.cache.access;
+    cache[name] = cache[name] || {};
 
-  // check for cached result
-  if (cache[name][pkg] !== undefined) {
-    return cache[name][pkg];
-  }
+    // check for cached result
+    if (cache[name][pkg] !== undefined) {
+      return cache[name][pkg];
+    }
 
-  // find all the branches that match the origin
-  var branches = findBranchesByPath({
-    dependencies: this.tree
-  }, pkg);
+    // find all the branches that match the origin
+    var branches = findBranchesByPath({
+      dependencies: this.tree
+    }, pkg);
 
-  var canAccess = branches.some(function(branch) {
-    // if the reference is to itself (branch.name)
-    // OR it's an immediate dependency (branch.dependencies[name])
-    if (branch.name === name || branch.dependencies && branch.dependencies[name]) {
-      return true;
-    }
-  });
+    var canAccess = branches.some(function(branch) {
+      // if the reference is to itself (branch.name)
+      // OR it's an immediate dependency (branch.dependencies[name])
+      if (branch.name === name || branch.dependencies && branch.dependencies[name]) {
+        return true;
+      }
+    });
 
-  debug.import(
-    "%s can%s be imported from %s",
-    name,
-    (canAccess ? "" : "not"),
-    origin
-  );
+    debug.import(
+      "%s can%s be imported from %s",
+      name,
+      (canAccess ? "" : "not"),
+      origin
+    );
+
+    // cache it for future lookups
+    cache[name][pkg] = canAccess;
+
+    return canAccess;
+  }.bind(this);
 
-  // cache it for future lookups
-  cache[name][pkg] = canAccess;
+  // check if we can access from the origin...
+  var canAccess = canAccessFrom(origin);
+
+  // if not...
+  if (!canAccess) {
+    // check for a symlink...
+    var realOrigin = fs.realpathSync(origin);
```

Comment:
is there a reason not to always `realpathSync` ? node's resolution algorithm takes that approach.

Reply:
This could make a file that is linked to appear like it belongs to a particular module no longer appear as part of the module. It's possible that our validations are being done at the wrong time against the wrong values and that fixing the way those work would be the best solution -- but I think using realpath all the time implies more testing and code changes.

Reply:
> but I think using realpath all the time implies more testing and code changes.

Makes sense.  I suspect this may be something todo, as it (from how I understand eyeglass to work) work more like the node `require`, and may lead to less surprises when symlinking. it's also possible I don't quite understand this part of eyeglass, and this doesn't make sense.